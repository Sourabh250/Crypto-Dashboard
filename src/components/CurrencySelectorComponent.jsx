import React,{ useState } from "react";
import { IoTriangle } from "react-icons/io5";
import { useDispatch, useSelector } from "react-redux";

function CurrencySelectorComponent() {
    const [list, setList] = useState(false);
    const dispatcher = useDispatch();
    const reduxData = useSelector(state => state.currency);

    // Function to handle currency selection
    const handleCurrency = (e) => {
        dispatcher({type: "SET_CURRENCY", payload: e.target.textContent});
        setList(false); // Close the currency list dropdown
    };

    const currencyList = ["INR","USD","EUR","JPY","GBP"]; // List of currency

    return (
        <div className="w-28 bg-gray-100 rounded-lg shadow-lg relative">
            <button className="w-full h-full flex justify-between items-center gap-1  p-4" onClick={() => setList(!list)}>{reduxData}<IoTriangle style={{transform: "rotate(180deg)"}} /></button>

            {/* Currency list */}
            {list && <ul className="absolute z-10 bg-gray-300 w-32 mt-2 rounded max-h-20 overflow-y-auto p-2 custom-scrollbar">
                {currencyList.map(currency => (
                    <li key={currency} className="p-1 hover:bg-red-200 cursor-pointer" onClick={handleCurrency}>{currency}</li>
                ))}
            </ul>}
        </div>
    )
}

export default CurrencySelectorComponent;