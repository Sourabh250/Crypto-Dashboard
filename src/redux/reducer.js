function reducer(state = {currency : "USD", selectedCurrencies: ["Ethereum"]},action) {
    switch(action.type) {
        // Update the selected currency in the state
        case "SET_CURRENCY":
            return {
                ...state,
                currency: action.payload
            };
        // Add a new cryptocurrency to the list of selected cryptocurrencies
        case "ADD_CRYPTO_CURRENCY" :
            return {
                ...state,
                selectedCurrencies: [...state.selectedCurrencies, action.payload],
            };
        // Remove a cryptocurrency from the list of selected cryptocurrencies
        case "REMOVE_CRYPTO_CURRENCY" :
            return {
                ...state,
                selectedCurrencies: state.selectedCurrencies.filter(data => data !== action.payload),
            };
        // Modify the list of selected cryptocurrencies to only contain the specified cryptocurrency
        case "MODIFY_CRYPTO_CURRENCY" :
            return {
                ...state,
                selectedCurrencies: [action.payload],
            }
        // Return the current state if the action type is not recognized
        default:
            return state;
    }
}

export default reducer;