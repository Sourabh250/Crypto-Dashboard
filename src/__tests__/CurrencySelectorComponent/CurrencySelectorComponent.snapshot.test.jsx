import React from "react";
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from '../../redux/reducer';
import CurrencySelectorComponent from '../../components/CurrencySelectorComponent';

const store = createStore(reducer);

test('CurrencySelectorComponent snapshot', () => {
    const tree = renderer.create(
        <Provider store={store}>
            <CurrencySelectorComponent />
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});