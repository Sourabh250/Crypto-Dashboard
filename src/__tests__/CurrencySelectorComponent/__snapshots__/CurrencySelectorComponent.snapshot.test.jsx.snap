// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencySelectorComponent snapshot 1`] = `
<div
  className="w-28 bg-gray-100 rounded-lg shadow-lg relative"
>
  <button
    className="w-full h-full flex justify-between items-center gap-1  p-4"
    onClick={[Function]}
  >
    USD
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
          "transform": "rotate(180deg)",
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M464 464H48a16 16 0 0 1-14.07-23.62l208-384a16 16 0 0 1 28.14 0l208 384A16 16 0 0 1 464 464z"
      />
    </svg>
  </button>
</div>
`;
